src/kbc_modules/cc_kbc/crypto/mod.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use aes_gcm::aead::{Aead, NewAead};
+use aes_gcm::Aes256Gcm;
+use anyhow::*;
+use rsa::pkcs8::EncodePublicKey;
+use rsa::{PaddingScheme, RsaPrivateKey, RsaPublicKey};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha384};
+
+const RSA_ALGORITHM: &str = "rsa-pkcs1v15";
+const RSA_PUBKEY_LENGTH: usize = 2048;
+
+const AES_GCM_256_ALGORITHM: &str = "aes-gcm-256";
+
+// The key inside TEE to decrypt confidential datas.
+#[derive(Debug, Clone)]
+pub struct TeeKey {
+    private_key: RsaPrivateKey,
+    public_key: RsaPublicKey,
+}
+
+impl TeeKey {
+    pub fn new() -> Result<TeeKey> {
+        let mut rng = rand::thread_rng();
+
+        let private_key = RsaPrivateKey::new(&mut rng, RSA_PUBKEY_LENGTH)?;
+        let public_key = RsaPublicKey::from(&private_key);
+
+        Ok(TeeKey {
+            private_key,
+            public_key,
+        })
+    }
+
+    // Export TEE public key as specific structure.
+    pub fn export_pubkey(&self) -> Result<TeePubKey> {
+        let pem_line_ending = rsa::pkcs1::LineEnding::default();
+        let pubkey_pem_string = self.public_key.to_public_key_pem(pem_line_ending)?;
+
+        Ok(TeePubKey {
+            algorithm: RSA_ALGORITHM.to_string(),
+            pubkey_length: RSA_PUBKEY_LENGTH.to_string(),
```

Comment:
curious: Why is this value a string? (I saw this is how it is defined in the [protocol doc](https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#attestation))

Reply:
Ah, this may be an oversight when writing KBS protocol documents. It should be more appropriate to use integer types here